worldlimiter.lua

```diff
@@ -1,29 +1,54 @@
+local WorldLimiter_Flag = false -- True when teleportation is about to occur, false otherwise
+local WorldLimiter_LastMessage = -100 -- The last time the player was sent a message about reaching the border
 function OnPlayerMoving( Player )
-	local LimitWorldWidth = WorldsWorldLimit[Player:GetWorld():GetName()]
+	local LimitChunks = WorldsWorldLimit[Player:GetWorld():GetName()]
 	
 	-- The world probably was created by an external plugin. Lets load the settings.
-	if not LimitWorldWidth then
+	if not LimitChunks then
 		LoadWorldSettings(Player:GetWorld())
-		LimitWorldWidth = WorldsWorldLimit[Player:GetWorld():GetName()]
+		LimitChunks = WorldsWorldLimit[Player:GetWorld():GetName()]
 	end
-	
-	if LimitWorldWidth > 0 then
+
+	if LimitChunks > 0 and WorldLimiter_Flag == false then
 		local World = Player:GetWorld()
-		local SpawnX = math.floor(World:GetSpawnX() / 16)
-		local SpawnZ = math.floor(World:GetSpawnZ() / 16)
-		local X = math.floor(Player:GetPosX() / 16)
-		local Z = math.floor(Player:GetPosZ() / 16)
-		if ( (SpawnX + LimitWorldWidth - 1) < X ) then 
-			Player:TeleportToCoords(Player:GetPosX() - 1, Player:GetPosY(), Player:GetPosZ()) 
-		end
-		if ( (SpawnX - LimitWorldWidth + 1) > X ) then
-			Player:TeleportToCoords(Player:GetPosX() + 1, Player:GetPosY(), Player:GetPosZ()) 
-		end
-		if ( (SpawnZ + LimitWorldWidth - 1) < Z ) then 
-			Player:TeleportToCoords(Player:GetPosX(), Player:GetPosY(), Player:GetPosZ() - 1) 
+		local Limit = LimitChunks * 16 - 1
+		local SpawnX = math.floor(World:GetSpawnX())
+		local SpawnZ = math.floor(World:GetSpawnZ())
+		local X = math.floor(Player:GetPosX())
+		local Z = math.floor(Player:GetPosZ())
+		local NewX = X
+		local NewZ = Z
+		
+		if X > SpawnX + Limit then 
```

Comment:
I'd prefer to have braces around `SpawnX + Limit` just to make it more clear, but I don't know if it's required in the Core.